### PR TITLE
Sign of life fix

### PR DESCRIPTION
**Pull request by krystian-hebel**

### CODE REVIEW

File: src/mainboard/pcengines/apu2/mainboard.c [hunk 1]
```diff
@@ -43,6 +43,9 @@
 #define SEC_REG_SERIAL_ADDR 0x1000
 #define MAX_SERIAL_LEN	    10
 
+void change_build_date_format(void);
+extern char coreboot_dmi_date[];
```
Comment (pietrushnic) [inline src/mainboard/pcengines/apu2/mainboard.c]: @krystian-hebel I'm not sure and where expose that?

File: src/mainboard/pcengines/apu2/mainboard.c [hunk 2]
```diff
@@ -359,3 +363,13 @@ struct chip_operations mainboard_ops = {
...
 	.final = mainboard_final
 };
+
+void change_build_date_format()
+{
+	char tmp[15];
+	strcpy(tmp, coreboot_dmi_date);
+	strncpy(coreboot_dmi_date,   tmp+6, 4);
+	strncpy(coreboot_dmi_date+4, tmp+3, 2);
+	strncpy(coreboot_dmi_date+6, tmp,   2);
```
Comment (pietrushnic) [inline src/mainboard/pcengines/apu2/mainboard.c]: @krystian-hebel this looks like magic dance. We should have some comment here what we do and why.

File: util/genbuild_h/genbuild_h.sh [hunk 3]
```diff
@@ -71,7 +71,7 @@ printf "#define COREBOOT_BUILD_YEAR_BCD 0x$(our_date "$DATE" +%y)\n"
 printf "#define COREBOOT_BUILD_MONTH_BCD 0x$(our_date "$DATE" +%m)\n"
 printf "#define COREBOOT_BUILD_DAY_BCD 0x$(our_date "$DATE" +%d)\n"
 printf "#define COREBOOT_BUILD_WEEKDAY_BCD 0x$(our_date "$DATE" +%w)\n"
-printf "#define COREBOOT_DMI_DATE \"$(our_date "$DATE" +%Y/%m/%d)\"\n"
+printf "#define COREBOOT_DMI_DATE \"$(our_date "$DATE" +%m/%d/%Y)\"\n"
```
Comment (pietrushnic) [inline util/genbuild_h/genbuild_h.sh]: @krystian-hebel can I assume this was reverted to be compatible with mainline?

File: src/mainboard/pcengines/apu2/mainboard.c (hunk 2 above)
Comment (krystian-hebel) [reply src/mainboard/pcengines/apu2/mainboard.c]: Done.

**Comment by miczyg1** [comment]
@krystian-hebel You should also do the same for apu1

**Comment by miczyg1** [comment]
@krystian-hebel we need the same functionality for apu1

File: src/mainboard/pcengines/apu1/romstage.c [hunk 4]
```diff
@@ -71,11 +75,20 @@ void board_BeforeAgesa(struct sysinfo *cb)
 	bool scon = check_console();
 
 	if(scon) {
-		// sign of life strings
-		printk(BIOS_ALERT, CONFIG_MAINBOARD_VENDOR " "
-			                   CONFIG_MAINBOARD_PART_NUMBER "\n");
-		printk(BIOS_ALERT, "coreboot build %s\n", COREBOOT_DMI_DATE);
-		printk(BIOS_ALERT, "BIOS version %s\n", COREBOOT_ORIGIN_GIT_TAG);
+        print_sign_of_life();
```
Comment (miczyg1) [inline src/mainboard/pcengines/apu1/romstage.c]: @krystian-hebel indentation

**Comment by miczyg1** [comment]
Everything works as expected. @pietrushnic any comments?

File: src/mainboard/pcengines/apu1/mainboard.c [hunk 5]
```diff
@@ -329,3 +333,18 @@ struct chip_operations mainboard_ops = {
...
 	.final = mainboard_final,
 };
+
+static void change_build_date_format()
+{
+	char tmp[15];
+    
+/* 
+ * Following lines change format of date in coreboot_dmi_date[]
+ * from "dd/mm/yyyy" to "yyyymmdd", as is expected by Pascal.
```
Comment (pietrushnic) [inline src/mainboard/pcengines/apu1/mainboard.c]: @krystian-hebel no names calling, please. Better would be `vendor`

File: src/mainboard/pcengines/apu2/mainboard.c [hunk 6]
```diff
@@ -359,3 +363,18 @@ struct chip_operations mainboard_ops = {
...
 	.final = mainboard_final
 };
+
+static void change_build_date_format()
+{
+	char tmp[15];
+    
+/* 
+ * Following lines change format of date in coreboot_dmi_date[]
+ * from "dd/mm/yyyy" to "yyyymmdd", as is expected by Pascal.
```
Comment (pietrushnic) [inline src/mainboard/pcengines/apu2/mainboard.c]: @krystian-hebel same here

**Comment by miczyg1** [comment]
@pietrushnic issues seem to be fixed.

**Comment by pietrushnic** [comment]
@krystian-hebel I will merge it but please submit info about this modification to CHANGELOG.md 

@miczyg1 FYI

**Pull request merged**